common/lanelet2_extension/CMakeLists.txt

```diff
@@ -43,21 +48,47 @@ set(deps_using_buildtool
   hardcoded_params
 )
 
+set(cpp_files
+  lib/autoware_osm_parser.cpp
+  lib/autoware_traffic_light.cpp
+  lib/local_frame_projector.cpp
+  lib/mgrs_projector.cpp
+  lib/CarmaUSTrafficRules.cpp
+  lib/CarmaTrafficLight.cpp
+  lib/PassingControlLine.cpp
+  lib/StopRule.cpp
+  lib/DigitalSpeedLimit.cpp
+  lib/DigitalMinimumGap.cpp
+  lib/RegionAccessRule.cpp
+  lib/DirectionOfTravel.cpp
+)
+
 
 # Extract environment variable value
-set(LANELET2_EXTENSION_LOGGER_TYPE $ENV{LANELET2_EXTENSION_LOGGER_TYPE})
-set(ROS_VERSION $ENV{ROS_VERSION})
+if(DEFINED ENV{LANELET2_EXTENSION_LOGGER_TYPE})
+  set(LANELET2_EXTENSION_LOGGER_TYPE $ENV{LANELET2_EXTENSION_LOGGER_TYPE})
+else()
+  set(LANELET2_EXTENSION_LOGGER_TYPE 0)
+endif()
+
+if(DEFINED ENV{ROS_VERSION})
+  set(ROS_VERSION $ENV{ROS_VERSION})
+else()
+  set(ROS_VERSION 0)
+endif()
+
 
-if(LANELET2_EXTENSION_LOGGER_TYPE EQUAL 1 )
 
-  message("lanelet2_extension using std::cout and cerr for logging")
+if(${LANELET2_EXTENSION_LOGGER_TYPE} EQUAL 1 )
+
+  message("lanelet2_extension using std::cout and cerr for logging ")
   add_definitions(-DLANELET2_EXTENSION_LOGGER_TYPE=1)
 
   set(catkin_depends_deps
     ${deps_using_buildtool}
   )
 
-elseif(ROS_VERSION EQUAL 2)
+elseif(${ROS_VERSION} EQUAL 2)
```

Comment:
In the loop on line 74, we are defining ROS_VERSION as dependent on whether the environment variable ${ROS_VERSION} is available or not, shouldn't we be using the same here?

Reply:
This is the pattern we have been using for all cmake variables. I don't think it was causing issues before, but I'm trying to keep things uniform.